ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -0,0 +1,200 @@
+        assert len(metadata_docs)==1
+        meta = {self.WORK_ID_FIELD:work_id, **metadata_docs[0]}
+
+        # find the pages docs
+        page_docs = [d for d in docs if d["item_type"] == "page"]
+
+        # transform into new dictionary with keys in `self.PAGE_ID_FIELD` and `self.OUTPUT_DOC_FIELDS`
+        pages = [self._transform_doc(doc,meta) for doc in page_docs]
+
+        # make sure sorted by numeric page num (i.e. "digital")
+        pages.sort(key=lambda page: page[self.PAGE_SORT_FIELD])
+        return meta, pages
+
+    def _transform_doc(self,doc:dict,meta:dict) -> dict:
+        """Reformat document dictionary"""
+
+        # get new dictionary
+        odoc={
+            key_new:doc.get(key_orig,'')
+            for key_new,key_orig in (
+                self.OUTPUT_DOC_FIELDS.items()
+            )
+        }
+
+        # return with page id
+        return {
+            self.PAGE_ID_FIELD:f'{meta[self.WORK_ID_FIELD]}_{odoc[self.PAGE_NUM_FIELD]}',
+            **odoc
+        }
+    
+    def _save_doc(self,doc_id:str) -> Tuple[str,dict]:
+        """Save document pages as json and return filename along with document's metadata"""
+
+        # get metadata and pages for this doc
+        meta,pages = self._get_meta_pages(doc_id)
+
+        # if pages, save json
+        if pages:
+            filename = os.path.join(self.path_texts, meta[self.WORK_ID_FIELD]+'.json')
+            os.makedirs(self.path_texts,exist_ok=True)
+            with open(filename,'w') as of:
+                json.dump(pages, of, indent=2)
+        
+        # otherwise, returned filename is blank to indicate no file saved
+        else:
+            filename=''
+        
+        return filename,meta
+
+
+    def save(self):
+        """Save the generated corpus text and metadata to files on disk"""
+
+        # save docs and gather metadata
+        metadata=[]
+        pdesc='Saved text to'
+        pbar=tqdm(total=self.doc_count, desc=f'{pdesc}: ...')
+        for doc_id in self.doc_ids:
+            # get saved filename and found metadata for this document
+            fn,meta = self._save_doc(doc_id)
+
+            # if we saved, update progress bar desc
+            if fn: pbar.set_description(f'{pdesc}: {fn}')
+
+            # tick
+            pbar.update()
+
+            # add this doc's meta to metadata
+            metadata.append(meta)
+        pbar.close()
+
+        # save metadata csv
+        with open(self.path_metadata,'w') as of:
+            json.dump(metadata, of, indent=2)
+        print(f'Saved metadata to: {self.path_metadata}')
+
+
+
+
+
+@contextmanager
+def logging_disabled(highest_level=logging.CRITICAL):
+    """Quick way to suppress solr logs as we iterate. Taken from https://gist.github.com/simon-weber/7853144"""
+    previous_level = logging.root.manager.disable
+    logging.disable(highest_level)
+    try: yield
+    finally: logging.disable(previous_level)
+
```

Comment:
This should be handled by updating django logging config rather than overriding here (we might need to turn logging on to debug something). What log level are you using and how noisy is it ?